src/dataset/composer.py

```diff
@@ -34,3 +34,45 @@ def __call__(self, image: np.ndarray, annotations: Dict) -> Any:
             image, annotations = func(image, annotations, **self.kwargs)
 
         return image, annotations
+
+
+def join_patches(patches: List[np.ndarray], filenames: List[str]) -> np.ndarray:
+    if len(patches) == 0:
+        raise ValueError("Input list of patches is empty.")
+
+    patches_positions = [x_y_from_filename(filename) for filename in filenames]
+    result_shape = calculate_result_shape_from_patches(patches, patches_positions)
+    result_image = np.zeros(result_shape, dtype=np.uint8)
+
+    for (y_at_result_image, x_at_result_image), patch in zip(patches_positions, patches):
+        height, width = patch.shape
+
+        result_image[
+            y_at_result_image:y_at_result_image + height,
+            x_at_result_image:x_at_result_image + width
+        ] = np.maximum(
+            result_image[y_at_result_image:y_at_result_image + height, x_at_result_image:x_at_result_image + width],
+            patch)
+
+    return result_image
+
+
+def calculate_result_shape_from_patches(patches: List[np.ndarray], dimensions: Tuple[int, int]) -> Tuple[int, int]:
+    max_x = 0
+    max_y = 0
+    for (x, y), patch in zip(dimensions, patches):
+        patch_height, patch_width = patch.shape
+        if x + patch_width > max_x:
+            max_x = x + patch_width
+        if y + patch_height > max_y:
+            max_y = y + patch_height
+
+    return (max_x, max_y)
+
+
+def x_y_from_filename(filename: str) -> Tuple[int, int]:
```

Comment:
What am I doing wrong? 
![Screenshot from 2023-12-14 12-27-41](https://github.com/laradru/ai-palatal-rugoscopy/assets/7481200/447fd241-138c-49d2-a43d-97903b0702a6)

Reply:
Mhmm... I was thinking that the typing module also provides options for primitive types. It's crazy because I used int throughout the entire code! :clap:

Reply:
Can we resolve this one?